openfasoc/generators/glayout/glayout/flow/blocks/composite/regulated_cascoded_current_mirror/current_mirror_WL_param.py

```diff
@@ -0,0 +1,126 @@
+from typing import Optional, Union
+
+from glayout.flow.pdk.util.snap_to_grid import component_snap_to_grid
+from pydantic import validate_arguments
+from glayout.flow.pdk.util.comp_utils import evaluate_bbox, prec_ref_center, movex, movey, to_decimal, to_float, move, align_comp_to_port, get_padding_points_cc
+from glayout.flow.pdk.util.port_utils import rename_ports_by_orientation, rename_ports_by_list, add_ports_perimeter, print_ports, set_port_orientation, rename_component_ports
+
+
+from current_mirror import current_mirror, current_mirror_netlist
+
+
+def sky130_add_current_mirror_labels(CMS: Component, transistor_type: str = "nfet",pdk: MappedPDK =sky130) -> Component:  # Re-introduce transistor_type
+	"""Add labels to the current mirror layout for LVS, handling both nfet and pfet."""
+
+	met2_pin = (69, 16)
+	met2_label = (69, 5)
+	met3_pin = (70, 16)
+	met3_label = (70, 5)
+
+
+
+	CMS.unlock()
+	move_info = []
+
+	# VREF label (for both gate and drain of transistor A, and dummy drains)
+	vref_label = rectangle(layer=met3_pin, size=(1, 1), centered=True).copy()
+	vref_label.add_label(text="VREF", layer=met3_label)
+	move_info.append((vref_label, CMS.ports["fet_A_gate_E"], None))  # Gate of A
+	move_info.append((vref_label, CMS.ports["fet_A_drain_E"], None)) # Drain of A
+
+
+	# VCOPY label (for drain of transistor B)
+	vcopy_label = rectangle(layer=met3_pin, size=(1, 1), centered=True).copy()
+	vcopy_label.add_label(text="VCOPY", layer=met3_label)
+	move_info.append((vcopy_label, CMS.ports["fet_B_drain_E"], None))  # Drain of B
+
+
+
+	# VSS/VDD label (for sources/bulk connection)
+	if transistor_type.lower() == "nfet":
+		bulk_net_name = "VSS"
+		bulk_pin_layer = met2_pin #met2 for nfet bulk
+		bulk_label_layer = met2_label #met2 for nfet bulk
+	else:  # pfet
+		bulk_net_name = "VDD"
+		bulk_pin_layer = met3_pin #met3 for pfet bulk
+		bulk_label_layer = met3_label #met3 for pfet bulk
+
+	bulk_label = rectangle(layer=bulk_pin_layer, size=(1, 1), centered=True).copy() #Layer changes based on type
+	bulk_label.add_label(text=bulk_net_name, layer=bulk_label_layer)
+	move_info.append((bulk_label, CMS.ports["fet_A_source_E"], None))  # Source of A
+	move_info.append((bulk_label, CMS.ports["fet_B_source_E"], None))  # Source of B
+
+	# VB label (connected to the dummy transistors' drains if present)
+	vb_label = rectangle(layer=met3_pin, size=(1, 1), centered=True).copy() #met3 for pfet
+	vb_label.add_label(text="VB" , layer=met3_label)
+	move_info.append((vb_label, CMS.ports["purposegndportscon_N"], None)) 
+	move_info.append((vb_label, CMS.ports["purposegndportscon_S"], None))
+
+	# Add labels to the component
+	for label, port, alignment in move_info:
+		if port:
+			alignment = ('c', 'b') if alignment is None else alignment
+			aligned_label = align_comp_to_port(label, port, alignment=alignment)
+			CMS.add(aligned_label)
+
+	return CMS.flatten()
+
+
+comp = current_mirror(sky130, numcols=2, device='nfet')
```

Comment:
It would be better to add these commands to a function or under `if __name__ == "__main__"` so it doesn't accidentally run as a side-effect of importing the file.